src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -0,0 +1,263 @@
+     */
+    public final boolean hasReads() {
+        return readsSource != null;
+    }
+
+    /**
+     * Are sources of intervals available?
+     *
+     * @return true if intervals are available, otherwise false
+     */
+    public final boolean hasIntervals() {
+        return intervals != null;
+    }
+
+    /**
+     * Window function that controls how much reference context to return for each read when
+     * using the reference source returned by {@link #getReference}. Tools should override
+     * as appropriate. The default function is the identity function (ie., return exactly
+     * the reference bases that span each read).
+     *
+     * @return reference window function used to initialize the reference source
+     */
+    public SerializableFunction<GATKRead, SimpleInterval> getReferenceWindowFunction() {
+        return RefWindowFunctions.IDENTITY_FUNCTION;
+    }
+
+    /**
+     * Returns the "best available" sequence dictionary. This will be the reference sequence dictionary if
+     * there is a reference, otherwise it will be the sequence dictionary constructed from the reads if
+     * there are reads, otherwise it will be null.
+     *
+     * TODO: check interval file(s) as well for a sequence dictionary
+     *
+     * @return best available sequence dictionary given our inputs
+     */
+    public SAMSequenceDictionary getBestAvailableSequenceDictionary() {
+        return hasReference() ? referenceDictionary : (hasReads() ? readsHeader.getSequenceDictionary() : null);
+    }
+
+    /**
+     * @return sequence dictionary for the reference, or null if there is no reference available
+     */
+    public SAMSequenceDictionary getReferenceSequenceDictionary() {
+        return referenceDictionary;
+    }
+
+    /**
+     * @return header for the reads, or null if there are no reads available
+     */
+    public SAMFileHeader getHeaderForReads() {
+        return readsHeader;
+    }
+
+    /**
+     * Loads the reads into a {@link JavaRDD} using the intervals specified. If no intervals
+     * were specified, returns all the reads (both mapped and unmapped).
+     *
+     * @return all reads from our reads input(s) as a {@link JavaRDD}, bounded by intervals if specified.
+     */
+    public JavaRDD<GATKRead> getReads() {
+        // If no intervals were specified (intervals == null), this will return all reads (mapped and unmapped)
+        return readsSource.getParallelReads(readInput, intervals);
+    }
+
+    /**
+     * @return our reference source, or null if no reference is present
+     */
+    public ReferenceDataflowSource getReference() {
+        return referenceSource;
+    }
+
+    /**
+     * @return our intervals, or null if no intervals were specified
+     */
+    public List<SimpleInterval> getIntervals() {
+        return intervals;
+    }
+
+    @Override
+    protected void runPipeline( JavaSparkContext sparkContext ) {
+        this.sparkContext = sparkContext;
```

Comment:
It doesn't have to be, but since most methods in the class require it, it seemed better to make it a member and save the extra param everywhere.